### ISSUE TITLE
Connected UDP, send fails 

### ISSUE THREAD

**Issue by evgasha** [libuv/libuv#3209]
I am using lib uv to implement UDP client that works in UDP connected mode, it means, I call 'uv_udp_connect' for my UDP handle, in case 'uv_udp_connect' return 'EINPROGRESS' , I start polling on UDP handle. When UDP handle enters into writable state , I make 'uv_udp_send' that fails with error code -39 (destination address required). After investigation of lib uv code, it clear that UDP handle flags property has missing UV_HANDLE_UDP_CONNECTED flag. As I understand it missing handling of EINPROGRESS error case in file libuv/src/unix/udp.c in method 'uv__udp_connect'.

* **Version**: 1.41.0
* **Platform**: MAC OS X Catalina

**Comment by santigimeno** [comment]
You're right, that cases was not considered. Do you have a reduced test case that reproduces the issue? Thanks

**Comment by evgasha** [comment]
Hi, my configuration is the following:

server: nc -l -p <port>

my test client:
udp_client   -use_connected_udp -port=<port> -ip=<ip>  
(code attached)
[main.cpp.zip](https://github.com/libuv/libuv/files/6662688/main.cpp.zip)



In the middle my Transparent Proxy network extension for Mac, and the issue happens at side of my test UDP client when my network extension is active. But after small fix in uvlib source code everything works fine.

**Comment by santigimeno** [comment]
Thanks for the info.
If you have a patch that fixes the issue it'd be great if you could open a PR with the fix.

**Comment by evgasha** [comment]
I created PR, fork evgasha/libuv, the fix only in unix leaf, I suggest in windows will be the same behavior, but I have no windows environment to test and fix it, sorry.

**Comment by evgasha** [comment]
Hi guys, what happen, I did changes by  @santigimeno suggestion, please review

**Comment by bnoordhuis** [comment]
@evgasha could it be you never actually opened a pull request? I can't find it, only a commit in your fork.

**Comment by juanarbol** [comment]
I can work on this one.

**Comment by bnoordhuis** [comment]
In 3.5 years we've had only this single report so that either means no one uses connected UDP, or that this was an issue local to the reporter's system.

I'm closing it out but let me know if there is reason to reopen.
